test/chat_page_test.dart

```diff
@@ -0,0 +1,140 @@
+import 'package:adventures_in_chat_app/chat_page.dart';
+import 'package:adventures_in_chat_app/models/conversation_item.dart';
+import 'package:adventures_in_chat_app/models/message.dart';
+import 'package:adventures_in_chat_app/services/api.dart';
+import 'package:adventures_in_chat_app/widgets/chat_message.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+
+class MockApi extends Mock implements Api {
+  MockApi(CollectionGet collectionGet);
+}
+
```

Comment:
What a magnificent set of tests!  My hat goes off to you sir!  I will try and live up to the example you have set!